src/control/server/mgmt_system.go

```diff
@@ -707,65 +709,110 @@ func (svc *mgmtSvc) SystemStart(ctx context.Context, pbReq *mgmtpb.SystemStartRe
+	if err != nil {
+		return errors.Wrap(err, "unable to determine path to self")
+	}
+
+	go func() {
+		if pause {
+			time.Sleep(50 * time.Millisecond)
+		}
+		if err := unix.Exec(myPath, append([]string{myPath}, os.Args[1:]...), os.Environ()); err != nil {
+			svc.log.Error(errors.Wrap(err, "Exec() failed").Error())
+		}
+	}()
+
+	return nil
+}
+
+// SystemErase implements the gRPC handler for erasing system metadata.
+func (svc *mgmtSvc) SystemErase(ctx context.Context, pbReq *mgmtpb.SystemEraseReq) (*mgmtpb.SystemEraseResp, error) {
+	// At a minimum, ensure that this only runs on MS replicas.
+	if err := svc.checkReplicaRequest(pbReq); err != nil {
 		return nil, err
 	}
 
+	svc.log.Debug("Received SystemErase RPC")
+
+	// If this is called on a non-leader replica, just nuke the local
+	// instance of the database and restart.
+	if !svc.sysdb.IsLeader() {
+		svc.eraseAndRestart(false)
+	}
+
+	// On the leader, we should first tell all servers to prepare for
+	// reformat by wiping out their engine superblocks, etc.
 	fanResp, _, err := svc.rpcFanout(ctx, fanoutRequest{
 		Method: control.ResetFormatRanks,
-		Hosts:  pbReq.GetHosts(),
-		Ranks:  pbReq.GetRanks(),
 	}, false)
 	if err != nil {
 		return nil, err
 	}
 
-	pbResp := &mgmtpb.SystemResetFormatResp{
-		Absentranks: fanResp.AbsentRanks.String(),
-		Absenthosts: fanResp.AbsentHosts.String(),
+	for _, mr := range fanResp.Results {
+		svc.log.Debugf("member response: %#v", mr)
 	}
+
+	pbResp := new(mgmtpb.SystemEraseResp)
 	if err := convert.Types(fanResp.Results, &pbResp.Results); err != nil {
 		return nil, err
 	}
 	for _, result := range pbResp.Results {
 		result.Action = "reset format"
 	}
 
-	svc.log.Debugf("Responding to SystemResetFormat RPC: %+v", pbResp)
-
-	return pbResp, nil
-}
+	if fanResp.Results.Errors() != nil {
+		return pbResp, nil
+	}
 
-// ClusterEvent management service gRPC handler receives ClusterEvent requests
-// from control-plane instances attempting to notify the MS of a cluster event
-// in the DAOS system (this handler should only get called on the MS leader).
-func (svc *mgmtSvc) ClusterEvent(ctx context.Context, req *sharedpb.ClusterEventReq) (*sharedpb.ClusterEventResp, error) {
-	if err := svc.checkLeaderRequest(req); err != nil {
+	// Next, tell all of the replicas to lobotomize themselves and restart.
+	peers, err := svc.sysdb.PeerAddrs()
+	if err != nil {
 		return nil, err
 	}
+	for _, peer := range peers {
+		peerReq := new(control.SystemEraseReq)
+		peerReq.AddHost(peer.String())
 
-	// indicate to handler that event has been forwarded
-	resp, err := svc.events.HandleClusterEvent(req, true)
-	if err != nil {
-		return nil, errors.Wrapf(err, "handle cluster event %+v", req)
+		if _, err := control.SystemErase(ctx, svc.rpcClient, peerReq); err != nil {
+			if control.IsConnectionError(err) {
+				continue
+			}
+			return nil, err
+		}
 	}
 
-	return resp, nil
+	// Finally, take care of the leader on the way out.
```

Comment:
maybe "take care of (this) leader on the way out"

Reply:
There's only one leader.